crates/guest/src/allocation.rs

```diff
@@ -71,41 +37,29 @@ pub fn consume_bytes(guest_ptr: GuestPtr) -> Vec<u8> {
             // must match the capacity set during the original allocation exactly
             len as usize,
         )
-    };
-
-    // This leads to an additional allocation for a new vector starting after the length prefix
-    // the old vector will be dropped and cleaned up by the allocator after this call
-    // the split off bytes will take ownership moving forward.
-    //
-    // Note that we could have tried to do something with std::slice::from_raw_parts() in this
-    // function but we'd still need a new allocation at the point of slice.to_vec() and then
-    // we'd need to manually free whatever the slice was pointing at.
-    v.split_off(std::mem::size_of::<Len>())
+    }
 }
 
-/// Attempt to write a slice of bytes into a length prefixed allocation.
+#[inline(always)]
```

Comment:
as a style point I think it's more readable to put attrs directly above the function rather than above the comment block, but not important